src/styles/blocks/header.scss

```
@@ -39,4 +39,10 @@
     font-size: 12px;
     font-weight: 500;
   }
+
+  &__link:hover {
+    transition: color;
+    transition-duration: $delayValue;
```

Add transition to origin element (not :hover) so the transisiton is smooth in both directions, you can use  short version:
```suggestion
    transition: color $delayValue
```